### DISCUSSION TITLE
Document inline next/script unmounts without id

### DISCUSSION THREAD

**Discussion by WillSquire** [vercel/next.js#28556] [Ideas]
### Describe the feature you'd like to request

Documentation that using next/script inline unmounts without `id` between page changes. This causes problems if initialising something only once. This is not quite clear: "Inline scripts wrapped with <Script> require an id attribute to be defined to track and optimize the script.".

### Describe the solution you'd like

As above.

### Describe alternatives you've considered

Trial and error...

**Comment by ivankopcik** [comment]
Same here, after adding `id` scripts are not working correctly. Trying to initiate hotjar and smartsupp.

To be clear, this is working correctly:

> <Script strategy={"lazyOnload"}>
>     {`(function(h,o,t,j,a,r){
>      h.hj=h.hj||function(){(h.hj.q=h.hj.q||[]).push(arguments)};
>      h._hjSettings={hjid: ... v=');}`
> </Script>

This is not working:

> <Script id={"hotjar"} strategy={"lazyOnload"}>
>         {`(function(h,o,t,j,a,r){
>           h.hj=h.hj||function(){(h.hj.q=h.hj.q||[]).push(arguments)};
>           h._hjSettings={hjid:XXXXX,hjsv:6};
>           a=o.getElementsByTagName('head')[0];
>           r=o.createElement('script');r.async=1;
>           r.src=t+h._hjSettings.hjid+j+h._hjSettings.hjsv;
>           a.appendChild(r);
>           })(window,document,'https://static.hotjar.com/c/hotjar-','.js?sv=');`}
> </Script>
